### PR TITLE
Replace special characters in names to avoid errors in generated Ballerina code

### PR DESCRIPTION
**Pull request by chathurace**

### CODE REVIEW

File: edi-tools/modules/x12xsd/x12xsd.bal [hunk 1]
```diff
@@ -31,10 +31,13 @@ public function convertFromX12XsdAndWrite(string inPath, string outPath) returns
 public function convertFromX12Xsd(xml x12xsd) returns edi:EdiSchema|error {
     xml elements = x12xsd/<xs:element>;
     xml root = elements[0];
-    string|error rootName = root.name;
-    if rootName is error {
-        return error("Root element name not found. " + root.toString(), rootName);
+    string rootName = "";
+    do {
+        rootName = check root.name;
+    } on fail var e {
```
Comment (anupama-pathirage) [inline edi-tools/modules/x12xsd/x12xsd.bal:37]:
```suggestion
    } on fail error e {
```

Comment (anupama-pathirage) [reply edi-tools/modules/x12xsd/x12xsd.bal:37]: This can be error right? And check for other places

File: edi-tools/modules/x12xsd/x12xsd.bal [hunk 2]
```diff
@@ -169,4 +182,13 @@ function getUnitElement(string name, xml x12xsd) returns xml|error {
...
+}
+
+public function getBalCompatibleName(string rawName) returns string {
+    string name = rawName.trim();
+    name = regex:replaceAll(name, "[^a-zA-Z1-9_]", "_");
+    if !regex:matches(name, "^[a-zA-Z].*") {
+        name = "A_" + name;
+    }
+    return name;
 }
```
Comment (LakshanWeerasinghe) [inline edi-tools/modules/x12xsd/x12xsd.bal:194]:
```suggestion
}

```

File: edi-tools/modules/x12xsd/x12xsd.bal [hunk 3]
```diff
@@ -169,4 +182,13 @@ function getUnitElement(string name, xml x12xsd) returns xml|error {
         return error("EDI segment/segment group not found in the input schema. Unit name: " + name);
     }
     return unitElement;
+}
+
+public function getBalCompatibleName(string rawName) returns string {
+    string name = rawName.trim();
+    name = regex:replaceAll(name, "[^a-zA-Z1-9_]", "_");
```
Comment (LakshanWeerasinghe) [inline edi-tools/modules/x12xsd/x12xsd.bal:189]: use the `regexp` langlib is better because the `regex` library is deprecated..

**Review by anupama-pathirage — APPROVED** [review]

File: edi-tools/modules/x12xsd/x12xsd.bal (hunk 3 above)
Comment (chathurace) [reply edi-tools/modules/x12xsd/x12xsd.bal:189]: Will update this in the next verison.

**Review by anupama-pathirage — APPROVED** [review]

**Pull request merged**